src/main/java/org/aksw/simba/lemming/ColouredGraphDecorator.java

```diff
@@ -0,0 +1,300 @@
+/**
+ * 
+ */
+package org.aksw.simba.lemming;
+
+import org.aksw.simba.lemming.mimicgraph.constraints.TripleBaseSingleID;
+import org.aksw.simba.lemming.util.IntSetUtil;
+
+import com.carrotsearch.hppc.BitSet;
+
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntSet;
+
+/**
+ * Base Decorator for Coloured Graph Class
+ * 
+ * @author Pranav
+ */
+public class ColouredGraphDecorator implements IColouredGraph {
+
+    protected IColouredGraph graph;
```

Comment:
We have a lot of variables that are just named `graph` :sweat_smile: 
Maybe `decorated` or `decoratedGraph` would be more precise :wink:

Reply:
Changed.